tests/metrics_test.go

```diff
@@ -153,14 +158,104 @@ var _ = Describe("Metrics", func() {
 		)
 	})
 
+	Context("SSP metrics", func() {
+		var originalVersion string
+
+		getDeployment := func(name string, namespace string) *apps.Deployment {
+			deployment := &apps.Deployment{}
+			err := apiClient.Get(ctx, client.ObjectKey{Name: name, Namespace: namespace}, deployment)
+			Expect(err).ToNot(HaveOccurred())
+			return deployment
+		}
+
+		updateVersion := func(version string) {
+			Eventually(func() error {
+				deployment := getDeployment(sspDeploymentName, sspDeploymentNamespace)
+
+				envs := deployment.Spec.Template.Spec.Containers[0].Env
+				var newEnvs []k8sv1.EnvVar
+
+				for _, env := range envs {
+					if env.Name != common.OperatorVersionKey {
+						newEnvs = append(newEnvs, env)
+					}
+				}
+
+				newEnvs = append(newEnvs, k8sv1.EnvVar{Name: common.OperatorVersionKey, Value: version})
+				deployment.Spec.Template.Spec.Containers[0].Env = newEnvs
+
+				return apiClient.Update(ctx, deployment)
+			}, 1*time.Minute, time.Second).ShouldNot(HaveOccurred())
+		}
+
+		waitForVersionUpdate := func(version string) {
+			Eventually(func() error {
```

Comment:
What about returning a bool instead of an error, to keep it simple?

Reply:
i think it makes it easier to understand to have the message from the error

Reply:
done

Reply:
i think it is clearer and more easier to understand in case of failure this way

Reply:
Then the error message is misleading, as it can not only occur when the RS was not found but also when the `rs.Status.ReadyReplicas == *rs.Spec.Replicas` was not met.

Reply:
message updated